src/jit/MemoryInl.h

```diff
@@ -173,6 +182,16 @@ void MemAddress::check(sljit_compiler* compiler, Operand* offsetOperand, sljit_u
             memArg.arg = SLJIT_MEM1(baseReg);
         }
 #endif /* SLJIT_32BIT_ARCHITECTURE */
+
+#if defined(ENABLE_EXTENDED_FEATURES)
+        if (size > 1 && (options & CheckNaturalAlignment)) {
```

Comment:
Maybe we should not set `CheckNaturalAlignment` when `size == 1`. An ASSERT can be added at the beginning of check.